Feliz.UseMediaQuery/UseMediaQuery.fs

```diff
@@ -38,26 +41,38 @@ let private makeQueries breakpoints =
 
 [<AutoOpen>]
 module UseMediaQueryExtension =
+     [<Emit("($0).addListener($1)")>]
+     let private addListener mqList fn = jsNative
+
+     [<Emit("($0).removeListener($1)")>]
+     let private removeListener mqList fn = jsNative
+
      type React with
-        /// A hook for media queries, this hook will force a component 
-        /// to re-render when the specified media query changes.  
-        static member useMediaQuery (query: string) =
-            let (mq, setMq) = React.useState(fun () -> window.matchMedia(query))
+        /// A hook for media queries, this hook will force a component
+        /// to re-render when the specified media query changes.
+        static member useMediaQuery (mediaQuery: string) =
+            let (mq, setMq) =
+                React.useState(fun _ -> window.matchMedia(mediaQuery).matches)
 
             React.useEffect(fun () ->
-                mq.addEventListener("change", fun e ->
-                    let mqEvent = unbox<MediaQueryList>(e)
-                    setMq mqEvent)
-                {new IDisposable with
-                     member this.Dispose() =
-                        mq.removeEventListener("change", fun _ -> ())}
+                let mediaQueryList = window.matchMedia(mediaQuery)
+                let handler = fun _ ->
+                        setMq(mediaQueryList.matches)
+
+                handler ()
+                addListener mediaQueryList handler
 
-            , [| query :> obj |])
+                let ret =
+                    {new IDisposable with
+                        member this.Dispose() =
+                            removeListener mediaQueryList handler}
+                ret
+            , [| setMq :> obj; mediaQuery :> obj |])
```

Comment:
I am not that `setMq` itself can be a dependency here and it doesn't really change

Reply:
Indeed. Fixed.